lib/optimizely/event/batch_event_processor.rb

```diff
@@ -102,7 +102,7 @@ def stop!
         @received.signal
       end
 
-      @is_started = false
+      @started = false
```

Comment:
What if `stop!` is called several times quickly? Would it be more correct this way, with checking `@started` first?

```ruby
def stop!
  return unless @started
  return unless @thread.alive? # not sure if this is still necessary?
  
  @logger.log(Logger::WARN, 'Stopping scheduler.')
  @started = false
  @mutex.synchronize do
    event_queue << SHUTDOWN_SIGNAL
    @received.signal
  end

  @thread.exit
end
```